apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
I just realized that ```Drafts.add``` is called only in one place in the app - ```Conversationiew.updateDraft```, maybe no need to do overcomplicated solution and just move ```Drafts.store``` call from ```Drafts.add```, and call it directly in ```ConversationView.updateDraft``` (still keeping simple promise-as-return-value in ```Drafts.store```)? :)